### ISSUE TITLE
Thanks for sharing. Could you provide additional code files? 

### ISSUE THREAD

**Issue by guyucowboy** [GarkP/LUNA16_Challange#1]
Hi, Görkem Polat
        Thanks for sharing. Could you provide additional code files?  The false positive nodules could be not reduced with these existing code files.
Best regards
Gu Yu

**Comment by GorkemP** [comment]
Hi,
tensorflow implementation was just to give an idea about implementation.
 I uploaded the full Keras implementation.

**Comment by guyucowboy** [comment]
Hi, Görkem Polat
Thanks for sharing your great work!  How to cite your work as reference style when writing a paper? 
Why fusing the result of the each model can obtain the better performance rather than the average performance?  
I may consult you when I have some questions.  Thanks for your help!
Best regards
Gu Yu

**Comment by GorkemP** [comment]
Hi,
In ML/DL algorithms, generally speaking, if the individual models make different types of errors, ensemble of the classifiers may provide better result than the average, even the best one. Building model on top of other models is called _Ensemble Learning_. For each instance/sample in the test set, minor errors can be eliminated by the other models. Think this concept as the _wisdom of the crowd_, even if some people make errors in decisions, decision of the majority determines the final result.

For the citation, you can use the following paper.
https://arxiv.org/abs/1811.01424

I uploaded the english version of it to the this repo.

**Comment by guyucowboy** [comment]
Hi, Görkem Polat
Thanks for your reply! -:)
Best regard
Gu Yu

**Comment by guyucowboy** [comment]
Hi, Görkem Polat
Thanks for your reply! -:) 
I have three question to consult you.
Which backend do you select? 
How to write the codes or set the parameters of the keras if I want to use multiple gpus?
Should I use the evaluationScript project in LUNA16 website to evaluate the false positive reduction result?
Thank you very much!
Best regard
Gu Yu

**Comment by GorkemP** [comment]
Hi,
I choosed tensorflow backend. I only used single GPU but tensorflow should easily manage multiple GPUs. You can download the evaluationScript and see the results yourself, challange does not accept new submissions anymore. evaluationScript is the script that is applied to the submitted results on the challenge website, so, you can objectively evaluate your results and compare with the submitted ones.

**Comment by guyucowboy** [comment]
Hi, Görkem Polat.
Thanks for your help! Consulting you can always obtain a lot of knowledge.-:) 
1. I found that you set CAND_THRESHOLD to early stop the training. Can the performance  be affected?
2. Why do you select Adadelta optimizer? How about other optimizers?
3. What is wrong with my model that rather low sensitivity (0.2)  was obtained at low fp rate(0.125,0.25) but sensitivity(0.9) at high fp rate (4,8) is OK?
Thank you very much!
Best regard
Gu Yu

**Comment by GorkemP** [comment]
Hi,

1) Yes, the performance can be affected. You should observe the cost of validation set and training set there can be overfitting. Indeed, my main aim to use early-stop is to finsh the training as soon as possible, I was a bit in a hurry :). I think, best thing is to implement an interactive function to observe costs throughout the training.

2) As far as I remember, it gave me the best result.

3) The main challenge in this study is having a high high recall rate in the lower false-positive rates. This is the hardest part. Generally algorithms give good results on the 4 or 8 FPs per scan.

**Comment by guyucowboy** [comment]
Hi, Görkem Polat.
Thanks for your help!  
Do you have other published papers and related the code projects？ I promise that I will cite them if they are used.
Best regards
Gu Yu

**Comment by GorkemP** [comment]
Hi, I just uploaded the my MSc thesis related to the topic. I advise you to look for google scholar, there are very recent papers related to the topic.

**Comment by guyucowboy** [comment]
Hi, Görkem Polat.
Where can I download your MSc thesis？ Thanks for your help!
Best regards
Gu Yu

**Comment by GorkemP** [comment]
Hi, I uploaded it to this repo.

**Comment by guyucowboy** [comment]
Hi, Görkem Polat.
Thanks for your helps！ I find your MSc thesis. Both of the paper and the thesis will be cited when I write my paper. And thank you for your valuable advice！
Best regards
Gu Yu

**Comment by GorkemP** [comment]
You are welcome, good luck in your work.
